hrabsence/CRM/HRAbsence/Upgrader.php

```diff
@@ -560,10 +560,22 @@ public function upgrade_1401() {
    * @see https://compucorp.atlassian.net/browse/PCHR-1892
    */
   public function upgrade_1402() {
-    $deleteQuery = "DELETE FROM civicrm_hrjobcontract_leave WHERE leave_type NOT IN (
-      SELECT id FROM civicrm_hrabsence_type
-    )";
-    CRM_Core_DAO::executeQuery($deleteQuery);
+    $absenceTypes = civicrm_api3('HRAbsenceType', 'get',
+      ['options' => ['limit' => 0], 'return' => ['name']]
+    );
+
+    if (empty($absenceTypes['values'])) {
+      return true;
+    }
+
+    $absenceTypeNames = array_column($absenceTypes['values'], 'name');
+
+    civicrm_api3('HRJobLeave', 'get', [
+      'leave_type' => ['NOT IN' => $absenceTypeNames],
```

Comment:
This doesn't look right. In leave_type we store the Absence Type ID, not the name